util/worker_test.go

```
@@ -0,0 +1,182 @@
+		})
+	}
+	errs := wp.Wait()
+	assert.Empty(t, errs, "Expected no errors, got: %v", errs)
+
+	finalCountAfterRestart := atomic.LoadInt32(&counter)
+	if finalCountAfterRestart != 8 {
+		t.Errorf("expected counter to be 8, got %d", finalCountAfterRestart)
+	}
+}
+
+func TestParallelSubmitsAndWaits(t *testing.T) {
```

golangci-lint (1.62.2)

TestParallelSubmitsAndWaits should use t.Cleanup instead of defer

(tparallel)